reference-implementation/__tests__/parsing-addresses.js

```diff
@@ -152,25 +144,20 @@ describe('Absolute URL addresses', () => {
       }`,
       'https://base.example/path1/path2/path3',
       {
-        about: [expect.toMatchURL('about:good')],
-        blob: [expect.toMatchURL('blob:good')],
-        data: [expect.toMatchURL('data:good')],
-        file: [expect.toMatchURL('file:///good')],
-        filesystem: [expect.toMatchURL('filesystem:good')],
-        http: [expect.toMatchURL('http://good/')],
-        https: [expect.toMatchURL('https://good/')],
-        ftp: [expect.toMatchURL('ftp://good/')],
-        import: [],
-        mailto: [],
-        javascript: [],
-        wss: []
+        about: ['about:good'],
+        blob: ['blob:good'],
+        data: ['data:good'],
+        file: ['file:///good'],
+        filesystem: ['filesystem:good'],
+        http: ['http://good/'],
+        https: ['https://good/'],
+        ftp: ['ftp://good/'],
+        import: ['import:bad'],
```

Comment:
The loss of these warnings is also troubling. These aren't even bare specifiers... they're invalid absolute URL specifiers. Why did we lose the ability to reject them?

Reply:
Absolute URL specifiers with non-fetch schemes are treated as non-URLs, so in this case the specifier `import:bad` would be successfully *parsed* but not allowed to be the result of resolution. In [a separate test](https://github.com/WICG/import-maps/blob/ccaf0c062738a2c1b93470ffffa297ed3f251b33/reference-implementation/__tests__/resolving.js#L94-L108), we show how non-fetch-scheme-URL module specifiers like this one can be made to map to resolvable URLs without error.

After addressing [your earlier comment about stripping bare specifiers](https://github.com/WICG/import-maps/pull/167/files#r314491076), these would end up removed with a warning at composition time.